### PR TITLE
Support on_crash & on_timeout callbacks for libafl_qemu modules

### PR DESCRIPTION
**Pull request by rmalmain**
mostly useful for drcov stuff for now.
should solve #1150

### CODE REVIEW

**Comment by domenukk** [comment]
Amazeballs

File: libafl_qemu/src/modules/drcov.rs [hunk 1]
```diff
@@ -322,6 +242,100 @@ where
...
+        _observers: &mut OT,
+        _exit_kind: &mut ExitKind,
+    ) where
+        OT: ObserversTuple<S::Input, S>,
+        ET: EmulatorModuleTuple<S>,
+    {
+        self.write();
+    }
+
+    fn on_crash(&mut self) {
```
Comment (domenukk) [inline libafl_qemu/src/modules/drcov.rs:332]: Only question is, should the callbacks have access to input, state etc.?

Comment (rmalmain) [reply libafl_qemu/src/modules/drcov.rs:332]: i started to implement this with these things as parameters, but i realized that for now it doesn't seem useful.
so i was planning to add these if it is needed later on

**Comment by tokatoka** [comment]
there should be a comment stating that "Never do something that could crash inside crash handler". I've at least seen twice when such mistake is made into libafl.

**Comment by rmalmain** [comment]
yes that's right, i'll add this

**Comment by rmalmain** [comment]
i'll even make them unsafe, it's easy to mess up in there

**Comment by domenukk** [comment]
Merge?

**Comment by domenukk** [comment]
@rmalmain  I'm still confused why this is different to.. the other..? crash handlers (forgot the name)

**Pull request merged**